### ISSUE TITLE
On inspect console, text on the right disappears when scrolling

### ISSUE THREAD

**Issue by Toranks** [wesnoth/wesnoth#7093]
### Game and System Information

The Battle for Wesnoth version 1.17.9 x86_64
Running on Microsoft Windows 7 Service Pack 1 (6.1.7601) x86_64
Distribution channel: Steam

Current video settings
======================

SDL video drivers:        [windows] dummy
Window size:              1920x1080
Game canvas size:         1920x1080
Final render target size: 1920x1080
Screen refresh rate:      60

### Description of the bug

When I try to scroll any section of inspect, the text disappears. That happens whether I use the mouse or the keyboard (arrows or page down). The only way I have to read it all is to use the copy button and read it with an external application. This did not happen in 1.16

### Steps to reproduce the behavior

1. Write :inspect (on debug mode)
2. Click on any section on the left
3. Try to scroll the text on the right if it fills more than the available space

### Expected behavior

It should allow me to scroll through all the text, like in 1.16

**Comment by Wedge009** [comment]
For clarity, inspection of game variables requires debug-mode.

I wonder if this is specific to how Windows renders things (maybe from the big graphics overhaul in 1.17.x?) because I could only confirm this on Windows, not Linux. The issue occurs in current master, both Windows 7 and 10.

**Comment by Wedge009** [comment]
The issue looks to have occurred between 1.17.5 and 1.17.6. I don't think I can do a bisect because during this period the vcpkg configuration was broken, relating to sdl2-mixer 'does not have required feature dynamic-load'.

**Comment by Toranks** [comment]
This happens on plugin manager too, from a certain amount of text
![image](https://user-images.githubusercontent.com/40789364/200313834-971a934d-9f94-4544-a7d7-31812f3a8f1f.png)

**Comment by Wedge009** [comment]
Yes, seems to be the same issue - I can only replicate on Windows, not Linux.

**Comment by Wedge009** [comment]
The add-ons issue cited on 7th November 2022 seems to be what @knyghtmare was reporting in #6879.

**Comment by Wedge009** [comment]
Just noting that while retesting #7201 on Linux, I also run into this.

**Comment by Pentarctagon** [comment]
@Wedge009 so this isn't Windows-specific, then?

**Comment by Wedge009** [comment]
I can only say I encounter the issue consistently in my Windows builds. My note on 27 August suggests I encountered it on a Linux build, but it's a much more rare occurrence. Reading my comment in #7201, it happened when I switched from a QHD screen to UHD (27-inch in both cases).

When I have time maybe I'll double-check on Linux. Perhaps it is related to the length of the text in some way.

**Comment by Wedge009** [comment]
Retesting master @ 60f58c96946 in Linux, inspecting units doesn't lose the text in the same way that my Windows build does. The issue is probably there somewhere, it just seems it happens less frequently/easily under Linux.

**Comment by Pentarctagon** [comment]
@CelticMinstrel @soliton- are you able to reproduce this on macOS? I've not seen it on Linux either.

**Comment by soliton-** [comment]
I've started a game of world conquest, opened inspect and looked at a unit. Scrolling down does not make the text disappear. At some point scrolling down the text looks weird though. Lines become compressed. At the top a line is about the size of my mouse cursor and at the bottom the cursor is clearly bigger.

**Comment by Pentarctagon** [comment]
So I guess it'd be related to what goes on with https://github.com/wesnoth/wesnoth/blob/2b4234afd74c4ecfc78612f9a6596e48dd39ef56/src/gui/widgets/scrollbar.cpp#L169

Not sure why `fire(event::NOTIFY_MODIFIED, *this, nullptr);` is commented out - the doc comment of `scroll()` doesn't seem to say anything about it despite the comment above that line of code.

**Comment by CelticMinstrel** [comment]
I vaguely remember at one point determining that it looked like an issue of the clipping rect being set incorrectly somewhere, but I don't remember if that's this same issue or a different one that had similar symptoms.

**Comment by Pentarctagon** [comment]
> Not sure why fire(event::NOTIFY_MODIFIED, *this, nullptr); is commented out - the doc comment of scroll() doesn't seem to say anything about it despite the comment above that line of code.

Turns out it does mean the comment literally in `scrollbar_base::scroll`:
```
/*
* @note I'm not sure if this should be in @ref set_item_position, but right now it
* causes weird viewport issues with listboxes. Selecting certain items makes the
* viewport to jump to the top, likely having to do with @ref recalculate dispatching
* an event (since it calls set_item_position). This seems like a safe spot for now.
*
* -- vultraz, 2017-09-13
*/
```

**Comment by Wedge009** [comment]
Seems resolved along with #7565. Please re-open if anyone finds it's still an issue.
